### PR TITLE
Fixed corner case with var_names

### LINKED ISSUE #334: minimization bug (Cannot read property '...' of undefined)
"Uncaught TypeError: Cannot read property 'x' of undefined" when running "webpack" in production mode

```js
(function(A) {
  (function() {
    doPrint();
  })();

  function doPrint() {
    print(A);
  }
}('Hello World!'));


function print(A) {
  if (!A.x) {
    console.log(A);
  }
}
```

Webpack configuration:
```js
module.exports = {
  mode: 'production',
  optimization: {
    minimize: true
  }
};
```

package.json:
```json

{
  "scripts": {
    "start": "webpack-dev-server --config=webpack.config.js"
  },
  "devDependencies": {
    "webpack": "4.30.0",
    "webpack-cli": "3.2.3",
    "webpack-dev-server": "3.1.14"
  }
}
```

When minification is enabled we get error
```bash
main.js:1 Uncaught TypeError: Cannot read property 'x' of undefined
    at main.js:1
```

image

As you can see this is wrong code because of variable `e` always undefined

```js
var e;
(e = e).x || console.log(e)
```

If turn off minification all work as expected

Sandbox repository with the bug - https://github.com/maksimr/playground/tree/webpack-minimize-bug

### PR DESCRIPTION
**Pull request by L2jLiga**
Fixes #334

### CODE REVIEW

**Comment by maksimr** [comment]
Ping. It would be nice to have this fix

File: lib/compress/index.js [hunk 1]
```diff
@@ -3982,7 +3982,7 @@ merge(Compressor.prototype, {
     AST_Scope.DEFMETHOD("var_names", function varNames() {
         var var_names = this._var_names;
         if (!var_names) {
-            this._var_names = var_names = this.parent_scope instanceof AST_Defun ? varNames.call(this.parent_scope) : new Set();
```
Comment (fabiosantoscode) [inline lib/compress/index.js]: Let's hope it's the last time we have to change this line in 2019 lol

**Comment by fabiosantoscode** [comment]
Thanks for the fix @L2jLiga

File: lib/compress/index.js (hunk 1 above)
Comment (L2jLiga) [reply lib/compress/index.js]: I hope too :)

**Pull request merged**